bridge/npbackend/src/_bh.c

```diff
@@ -1105,7 +1105,44 @@ static PyObject* BhArray_Repr(PyObject *self) {
         return NULL;
     }
 
-    PyObject *str = PyArray_Type.tp_repr(t);
+    PyObject *str = NULL;
+
+    BH_PyArrayObject *base = &((BhArray*) self)->base;
+    if (base->nd == 0) {
+        // 0-rank array -> single value
+        void *data = PyArray_GetPtr((PyArrayObject*) self, 0);
+        char c[32];
+
+        switch (base->descr->type) {
+            case 'i': // int32
+                snprintf(c, sizeof(c), "%d", *((npy_int*) data));
+                break;
+            case 'l': // int64
+                snprintf(c, sizeof(c), "%ld", *((npy_long*) data));
+                break;
+            case 'I': // uint32
+                 snprintf(c, sizeof(c), "%u", *((npy_uint*) data));
+                break;
+            case 'L': // uint64
+                 snprintf(c, sizeof(c), "%lu", *((npy_ulong*) data));
+                break;
+            case 'f': // float
+                snprintf(c, sizeof(c), "%.6g", *((npy_float*) data));
+                break;
+            case 'd': // double
+                snprintf(c, sizeof(c), "%.6g", *((npy_double*) data));
+                break;
+        }
+
+        if (c[0] == 0) {
+            str = PyArray_Type.tp_repr(t);
+        } else {
+            str = PyString_FromString(c);
```

Comment:
Python3 doesn't have `PyString_FromString()` use something like:
```c
#if PY_MAJOR_VERSION > 2
    return PyUnicode_FromString(c);
#else
    return PyString_FromString(c);
#endif
```

Reply:
I use the already defined `NPY_PY3K` in the above way.